packages/react/src/components/Callout/CalloutContent.base.tsx

```diff
@@ -111,6 +111,37 @@ function useBounds(
   return getBounds;
 }
 
+/**
+ * (Hook) to return the maximum available height for the Callout to render into.
+ */
+function useMaxHeight(
+  positions: ICalloutPositionedInfo | undefined,
+  { calloutMaxHeight, directionalHint, directionalHintFixed, hidden }: ICalloutProps,
+  getBounds: () => IRectangle | undefined,
+) {
+  const [maxHeight, setMaxHeight] = React.useState<number | undefined>();
+  const { top, bottom } = positions?.elementPosition ?? {};
+
+  React.useEffect(() => {
+    const isForcedInBounds = typeof top === 'number' && typeof bottom === 'number';
+    const { top: topBounds, bottom: bottomBounds } = getBounds() ?? {};
+
+    if (!calloutMaxHeight && !isForcedInBounds && !hidden) {
+      if (typeof top === 'number' && bottomBounds) {
+        setMaxHeight(bottomBounds - top);
+      } else if (typeof bottom === 'number' && typeof topBounds === 'number' && bottomBounds) {
+        setMaxHeight(bottomBounds - topBounds - bottom);
+      }
```

Comment:
Wouldn't we need one more case here for when the `else if` above isn't met either?

Reply:
The else case is that the maxHeight isn't changed at all. It should only be reached if either the positions or bounds aren't defined yet, and I thought it would be better to not change the callout's `maxHeight` value while e.g. the positions were still being calculated.

I could do an explicit
```js
else {
  // do nothing
}
```
if you think that would be better?

Reply:
I think it is fine to not do an extra else with nothing there then, maybe the best way to do this is to just add a comment after saying something like `// if these two conditions are not meant then we should not modify the callout's maxHeight value`.